### PR TITLE
fix resize_mask ignoring NEAREST_EXACT interpolation

### LINKED ISSUE #9188: InterpolationMode.NEAREST_EXACT and InterpolationMode.NEAREST are the same for tv_tensors.Mask
### 🐛 Describe the bug

When resizing tensor wrapped in tv_tensors.Mask interpolation mode NEAREST_EXACT is changed to NEAREST

```python
import torch
import torch.nn.functional as F
import torchvision.transforms.v2 as T
import torchvision.transforms.v2.functional as TF
from torchvision import tv_tensors


def compare(mask, size):
    mask_nearest = TF.resize(mask, size=size, interpolation=T.InterpolationMode.NEAREST)
    mask_nearest_exact = TF.resize(mask, size=size, interpolation=T.InterpolationMode.NEAREST_EXACT)
    mask_nearest_interpolate = F.interpolate(mask.unsqueeze(0), size=size, mode='nearest').squeeze(0)
    mask_nearest_exact_interpolate = F.interpolate(mask.unsqueeze(0), size=size, mode='nearest-exact').squeeze(0)
    
    print(torch.equal(mask_nearest, mask_nearest_exact))
    print(torch.equal(mask_nearest_interpolate, mask_nearest_exact_interpolate))
    print(torch.equal(mask_nearest, mask_nearest_interpolate))
    print(torch.equal(mask_nearest_exact, mask_nearest_exact_interpolate))
    

height = 64
width = 64
new_height = 16
new_width = 16
mask = torch.randint(low=0, high=2, size=(1, height, width)).float()

compare(mask, (new_height, new_width))
print()

mask = tv_tensors.Mask(mask)
compare(mask, (new_height, new_width))
```

```
False
False
True
True

True
False
True
False
```

### Versions

torch = "2.8.0"
torchvision = "0.23.0"

### PR DESCRIPTION
**Pull request by Mr-Neutr0n**
Fixes #9188

`resize_mask()` was hardcoding `InterpolationMode.NEAREST` and ignoring the user-specified interpolation mode entirely. This meant `NEAREST_EXACT` (which is the more correct interpolation — matching PIL and scikit-image behavior) got silently dropped when resizing a `tv_tensors.Mask`.

The fix adds an `interpolation` parameter to `resize_mask()` and passes it through from the dispatch function. Only `NEAREST` and `NEAREST_EXACT` are accepted — anything else (bilinear, bicubic, etc.) falls back to `NEAREST` since those don't make sense for masks.

### CODE REVIEW

**Comment by zy1git** [comment]
@Mr-Neutr0n Hi,

Thanks a lot for creating the PR. I am reviewing it. At the same time, could you please take a look at the test errors and address those?

**Comment by zy1git** [comment]
I took a look at the test errors and wanted to share a few observations:

a, the signature order does not match F.resize: resize_mask(mask, size, max_size, interpolation) should be resize_mask(mask, size, interpolation, max_size).
b, set literal {NEAREST, NEAREST_EXACT} is not compatible with TorchScript.  
c, The default value (NEAREST) differs from F.resize (BILINEAR) in the test file, which causes `test_functional_signature` to fail. 

If you have any questions while addressing the CI test failures, please feel free to reach out.

**Comment by Mr-Neutr0n** [comment]
@zy1git thank you for the three-point review, and sorry for going quiet on it — that was a concrete, actionable list and it deserved a faster response than five months.

**This is superseded and I'm closing it.** #9497 landed on 2026-05-26 and closed #9188.

Worth recording that the merged version implements your review exactly, on all three points:

| your note | merged in #9497 |
|---|---|
| a. signature order must match `F.resize` | `resize_mask(mask, size, interpolation, max_size)` ✓ |
| b. set literal isn't TorchScript-compatible | `if interpolation != NEAREST and interpolation != NEAREST_EXACT` ✓ |
| c. default must match `F.resize`'s `BILINEAR` | defaults to `BILINEAR`, then coerces to `NEAREST` with a comment explaining why ✓ |

It also went further than mine by routing through `_check_interpolation`, which picks up string support from #9461 for free — my version only handled `int` and the enum.

Verified against released `torchvision 0.28.0` rather than assuming, since my first attempt gave a false negative:

```python
mask = tv_tensors.Mask(torch.randint(0, 9, (1, 7, 11)))
size = [10, 13]

NEAREST vs NEAREST_EXACT differ?     True    <- #9188 fixed
string 'nearest-exact' == enum?      True
default coerces to NEAREST?          True
BILINEAR coerces to NEAREST?         True
```

The false negative is worth a note for anyone testing this later: I first checked with `size=[14, 22]` against a `(1, 7, 11)` mask, an exact 2× upsample where `nearest` and `nearest-exact` agree by construction. The two tensors matched and it looked like the bug was still live. The scale factor has to be non-integral for the algorithms to disagree at all.

Thanks also to @joannemiki57 for taking it over the line.
